Field.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace gameoflife;
+
+class Field
+{
+    protected $field = [];
+    protected $puffer = [];
+    protected $generationCount = 1;
+    protected $width;
+    protected $height;
+    protected $maxSteps;
+    protected $history = [];
+
+    /**
+     * Field constructor
+     * Constructs an empty field, which can be overwritten by the child constructor.
+     * @param $_width int
+     * @param $_height int
+     * @param $_maxSteps int
+     */
+    public function __construct($_width, $_height, $_maxSteps)
+    {
+        $this->width = $_width;
+        $this->height = $_height;
+        $this->maxSteps = $_maxSteps;
+        for ($y = 0; $y < $this->height; $y++) {
+            for ($x = 0; $x < $this->width; $x++) {
+                $this->field[$x][$y] = 0;
+            }
+        }
+    }
+
+    /**
+     * Print the field
+     * Generates a field and replaces 1->"$" and 0->" ".
+     */
+    public function printField()
+    {
+        for ($y = 0; $y < $this->height; $y++)
+        {
+            for ($x = 0; $x < $this->width; $x++)
+            {
+                echo($this->field[$x][$y] ? "$" : " ");
+            }
+            echo "\n";
+        }
+    }
+
+    /**
+     * Count living neighbours
+     * @param $_x int X-coordinates of the field.
+     * @param $_y int Y-coordinates of the field.
+     * @return int Of the neighbour-count.
+     */
+    function countNeighbours($_x, $_y)
+    {
+        $neighbours = 0;
+
+        for ($ny = $_y - 1; $ny <= $_y + 1; $ny++) {
+            for ($nx = $_x - 1; $nx <= $_x + 1; $nx++) {
+                if ($nx >= $this->width or $nx < 0) {
+                    continue;
+                }
+                if ($ny >= $this->height or $ny < 0) {
+                    continue;
+                }
+                if ($nx == $_x and $ny == $_y) {
+                    continue;
+                }
+
+                if ($this->field[$nx][$ny] == 1) {
+                    $neighbours++;
+                }
+            }
+        }
+        return $neighbours;
+    }
+
+    /**
+     * Calculate next generation.
+     * First $puffer is created, which copies the contents of the $fields variable and sets the cell-status to 0 (dead).
+     * After that, the "set-alive-conditions" will be implemented and cells fitting the conditions will be set to 1 (alive).
+     */
+    function nextGeneration()
+    {
+        //$puffer = [];
+
+        //set puffer to 0
+        for ($y = 0; $y < $this->height; $y++) {
+            for ($x = 0; $x < $this->width; $x++) {
+                $this->puffer[$x][$y] = 0;
```

Comment:
If you need `$this->puffer` only in this method, there is no need to make it a member-variable. You can just use it as local variable. Since in reality this is also a field, why not name it `$field` or more specifically `$nextField`?